pylint/checkers/classes/class_checker.py

```diff
@@ -1443,7 +1458,9 @@ def _check_redefined_slots(
                 node=slots_node,
             )
 
-    def _check_slots_elt(self, elt, node):
+    def _check_slots_elt(
+        self, elt: nodes.Attribute | nodes.Const, node: nodes.ClassDef
```

Comment:
Most of the values are based on what were the actual type of the node when launching the tests. There's often a lof of condition applied on things after the inference is done so it narrows the type of sub-functions quite a lot sometime.

Reply:
Yeah, but just based on the current typing and code this typing is incorrect. We should mostly rely on the code rather than on tests as the tests will never be cover every edge case.